### ISSUE TITLE
remote-ssh keeps downloading wrong architecture on arm64 destination host

### ISSUE THREAD

**Issue by ICHx** [microsoft/vscode-remote-release#4889]
Just now the source host vscode updated itself to 1.55.2, the target host is on arm64.

Version: 1.55.2 (system setup)
Commit: 3c4e3df9e89829dce27b7b5c24508306b151f30d
Date: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

Steps to Reproduce:

1. Connect to server
2. Server exit.

Does this issue occur when you try this locally?:No
Does this issue occur when you try this locally and all extensions are disabled?: No[remote-host issue]

I investigated into the .vscode-server directory, and found that the new VSCODE keep downloading the wrong architecture.

New version : 3c4e3df9e89829dce27b7b5c24508306b151f30d
![image](https://user-images.githubusercontent.com/11556527/114878493-df7d0d00-9e32-11eb-9dc5-b8932a84e0f9.png)

Old version: 08a217c4d27a02a5bcde898fd7981bda5b49391b
![image](https://user-images.githubusercontent.com/11556527/114878550-f1f74680-9e32-11eb-9190-2e2d9da89c61.png)

**Comment by ICHx** [comment]
Downgrading source host to v1.55.1 immediately got back to work.

**Comment by roblourens** [comment]
Please share the log from the Remote-SSH output channel for 1.55.2. This is driven by the extension, the vscode version shouldn't matter in theory

**Comment by ICHx** [comment]
[10:58:47.546] Log Level: 2
[10:58:47.546] remote-ssh@0.65.3
[10:58:47.546] darwin arm64
[10:58:47.547] SSH Resolver called for "ssh-remote+zuk", attempt 1
[10:58:47.547] "remote.SSH.useLocalServer": true
[10:58:47.547] "remote.SSH.path": undefined
[10:58:47.547] "remote.SSH.sshConfigurationFile": undefined
[10:58:47.547] "remote.SSH.useFlock": true
[10:58:47.547] "remote.SSH.lockfilesInTmp": false
[10:58:47.547] "remote.SSH.localServerDownload": auto
[10:58:47.547] "remote.SSH.remoteServerListenOnSocket": false
[10:58:47.547] "remote.SSH.showLoginTerminal": false
[10:58:47.548] "remote.SSH.defaultExtensions": []
[10:58:47.548] "remote.SSH.loglevel": 2
[10:58:47.548] SSH Resolver called for host: zuk
[10:58:47.548] Setting up SSH remote "zuk"
[10:58:47.549] Acquiring local install lock: /var/folders/4k/cbvl38v96jvfnqjntrxcywv40000gn/T/vscode-remote-ssh-zuk-install.lock
[10:58:47.552] Looking for existing server data file at /Users/rdevops/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-3c4e3df9e89829dce27b7b5c24508306b151f30d-0.65.3/data.json
[10:58:47.553] Using commit id "3c4e3df9e89829dce27b7b5c24508306b151f30d" and quality "stable" for server
[10:58:47.554] Install and start server if needed
[10:58:47.555] PATH: /usr/local/google-cloud-sdk/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/local/bin:/opt/local/sbin:/opt/homebrew/sbin
[10:58:47.555] Checking ssh with "ssh -V"
[10:58:47.559] > OpenSSH_8.1p1, LibreSSL 2.7.3

[10:58:47.561] askpass server listening on /var/folders/4k/cbvl38v96jvfnqjntrxcywv40000gn/T/vscode-ssh-askpass-eed4e14e804639db2765c865da735cfa8bf8d612.sock
[10:58:47.561] Spawning local server with {"ipcHandlePath":"/var/folders/4k/cbvl38v96jvfnqjntrxcywv40000gn/T/vscode-ssh-askpass-a348f4a67c8f210c4e6e84bf24e6ccd685a4e2d3.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","52618","-o","ConnectTimeout=15","zuk"],"dataFilePath":"/Users/rdevops/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-3c4e3df9e89829dce27b7b5c24508306b151f30d-0.65.3/data.json"}
[10:58:47.561] Local server env: {"DISPLAY":":0","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/rdevops/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.3/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)","VSCODE_SSH_ASKPASS_MAIN":"/Users/rdevops/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.3/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/4k/cbvl38v96jvfnqjntrxcywv40000gn/T/vscode-ssh-askpass-eed4e14e804639db2765c865da735cfa8bf8d612.sock"}
[10:58:47.561] Spawned 18515
[10:58:47.619] > local-server> Spawned ssh: 18524
[10:58:47.622] stderr> OpenSSH_8.1p1, LibreSSL 2.7.3
[10:58:48.051] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:2Op7xAXkYvXk9w0mP4jQBBY9s9QTqLRK51CuJ2azRmg
[10:58:48.267] stderr> Authenticated to 192.168.0.202 ([192.168.0.202]:8122).
[10:58:48.615] > Welcome to Ubuntu 20.04.2 LTS (GNU/Linux 4.4.205-LineageOS+ armv8l)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> Ubuntu 20.04 LTS [running via Linux Deploy]
[10:58:48.630] > ready: 57a9646d21e7
[10:58:48.646] > Linux 4.4.205-LineageOS+ #1 SMP PREEMPT Thu Apr 8 10:51:38 UTC 2021
[10:58:48.646] Platform: linux
[10:58:48.705] > 57a9646d21e7: running
[10:58:48.762] > Acquiring lock on /home/phone/.vscode-server/bin/3c4e3df9e89829dce27b7b5c24508306b151f30d/vscode-remote-lock.phone.3c4e3df9e89829dce27b7b5c24508306b151f30d
[10:58:48.779] > Installing to /home/phone/.vscode-server/bin/3c4e3df9e89829dce27b7b5c24508306b151f30d...
[10:58:48.791] > 57a9646d21e7%%1%%
> Downloading with wget
[10:59:00.297] > Download complete
[10:59:00.359] > 57a9646d21e7%%2%%
> tar --version:
[10:59:00.401] > tar (GNU tar) 1.30
> Copyright (C) 2017 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> 
> Written by John Gilmore and Jay Fenlason.
[10:59:01.598] > Checking /home/phone/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.log and /home/phone/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.pid for a running server
> Looking for server with pid: 31267
> Found running server...
>  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
> Checking server status on port 42158 with wget
> 57a9646d21e7: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==42158==
> osReleaseId==ubuntu==
> arch==armv8l==
> tmpDir==/tmp==
> platform==linux==
> unpackResult==success==
> didLocalDownload==0==
> downloadTime==11533==
> installTime==1101==
> extInstallTime====
> serverStartTime====
> connectionToken==a1aaa1a1-a1a1-1a1a-11a1-111a11111aa1==
> 57a9646d21e7: end
[10:59:01.598] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==42158==
osReleaseId==ubuntu==
arch==armv8l==
tmpDir==/tmp==
platform==linux==
unpackResult==success==
didLocalDownload==0==
downloadTime==11533==
installTime==1101==
extInstallTime====
serverStartTime====
connectionToken==a1aaa1a1-a1a1-1a1a-11a1-111a11111aa1==

[10:59:01.599] Remote server is listening on 42158
[10:59:01.599] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":42158},"osReleaseId":"ubuntu","arch":"armv8l","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/tmp","platform":"linux","connectionToken":"a1aaa1a1-a1a1-1a1a-11a1-111a11111aa1"},"downloadTime":11533,"installTime":1101,"installUnpackCode":"success"}
[10:59:01.600] Persisting server connection details to /Users/rdevops/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-3c4e3df9e89829dce27b7b5c24508306b151f30d-0.65.3/data.json
[10:59:01.601] Starting forwarding server. localPort 52626 -> socksPort 52618 -> remotePort 42158
[10:59:01.601] Forwarding server listening on 52626
[10:59:01.602] Waiting for ssh tunnel to be ready
[10:59:01.602] [Forwarding server 52626] Got connection 0
[10:59:01.603] Tunneled 42158 to local port 52626
[10:59:01.603] Resolved "ssh-remote+zuk" to "127.0.0.1:52626"
[10:59:01.611] ------




[10:59:01.657] [Forwarding server 52626] Got connection 1
[10:59:01.710] [Forwarding server 52626] Got connection 2

**Comment by ICHx** [comment]
> Please share the log from the Remote-SSH output channel for 1.55.2. This is driven by the extension, the vscode version shouldn't matter in theory

but changing back to v1.55.1 did successfully work again.

**Comment by ICHx** [comment]
The main issue is that the new VSCODE environment download arm32 on remote host, but old one did download arm64.  arm32 won't work on my environment.

**Comment by roblourens** [comment]
Are you sure it's wrong? The log shows the server was running so it started up at least. 

Could you run the command "Uninstall VS Code Server on Host" to remove it from the remote, then connect again to reinstall it, then share the log from that connection, so I can see the install logs?

**Comment by ICHx** [comment]
The server was not running, the client complains about unauthorized.  And the architecture being downloaded on remote is wrong.  I tried to remove .vscode-server/bin/*

**Comment by ICHx** [comment]
I had to downgrade all my clients [pc/mac] to 1.55.1 to successfully connect to that ssh remote server.

**Comment by roblourens** [comment]
I can't do anything without the log that includes installing the server

**Comment by ICHx** [comment]
Could you connect to a arm Linux remote server using vscode v1.55.2 successfully?
From multiple clients running v1.55.2 the same issue took place.  

I currently is not able to test v.155.2.  All my vscode installs are reverted.

**Comment by roblourens** [comment]
Other users are able to

**Comment by ICHx** [comment]
```
[11:43:10.571] Log Level: 2
[11:43:10.575] remote-ssh@0.65.4
[11:43:10.575] linux x64
[11:43:10.576] SSH Resolver called for "ssh-remote+zuk", attempt 1
[11:43:10.577] "remote.SSH.useLocalServer": true
[11:43:10.577] "remote.SSH.path": undefined
[11:43:10.578] "remote.SSH.configFile": undefined
[11:43:10.578] "remote.SSH.useFlock": true
[11:43:10.578] "remote.SSH.lockfilesInTmp": false
[11:43:10.581] "remote.SSH.localServerDownload": auto
[11:43:10.582] "remote.SSH.remoteServerListenOnSocket": false
[11:43:10.582] "remote.SSH.showLoginTerminal": false
[11:43:10.582] "remote.SSH.defaultExtensions": []
[11:43:10.582] "remote.SSH.loglevel": 2
[11:43:10.583] SSH Resolver called for host: zuk
[11:43:10.583] Setting up SSH remote "zuk"
[11:43:10.586] Acquiring local install lock: /tmp/vscode-remote-ssh-zuk-install.lock
[11:43:10.597] Looking for existing server data file at /home/ricts/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-3c4e3df9e89829dce27b7b5c24508306b151f30d-0.65.4/data.json
[11:43:10.598] Using commit id "3c4e3df9e89829dce27b7b5c24508306b151f30d" and quality "stable" for server
[11:43:10.601] Install and start server if needed
[11:43:10.605] PATH: /home/ricts/.local/bin:/home/ricts/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[11:43:10.606] Checking ssh with "ssh -V"
[11:43:10.622] > OpenSSH_8.2p1 Ubuntu-4ubuntu0.2, OpenSSL 1.1.1f  31 Mar 2020

[11:43:10.626] askpass server listening on /run/user/1000/vscode-ssh-askpass-a36aa95f9b47787495bb61a13c1cfe3d97d3b9c1.sock
[11:43:10.626] Spawning local server with {"ipcHandlePath":"/run/user/1000/vscode-ssh-askpass-72376cf74ff33c55032f13c31ddac73f7c687f1e.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","46245","-o","ConnectTimeout=15","zuk"],"dataFilePath":"/home/ricts/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-3c4e3df9e89829dce27b7b5c24508306b151f30d-0.65.4/data.json"}
[11:43:10.626] Local server env: {"DISPLAY":":0","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/usr/share/code/code","VSCODE_SSH_ASKPASS_MAIN":"/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/run/user/1000/vscode-ssh-askpass-a36aa95f9b47787495bb61a13c1cfe3d97d3b9c1.sock"}
[11:43:10.630] Spawned 10468
[11:43:10.733] > local-server> Spawned ssh: 10476
[11:43:10.735] stderr> OpenSSH_8.2p1 Ubuntu-4ubuntu0.2, OpenSSL 1.1.1f  31 Mar 2020
[11:43:13.788] stderr> ssh: connect to host 192.168.0.202 port 8122: No route to host
[11:43:13.791] > local-server> ssh child died, shutting down
[11:43:13.796] Local server exit: 0
[11:43:13.797] Received install output: OpenSSH_8.2p1 Ubuntu-4ubuntu0.2, OpenSSL 1.1.1f  31 Mar 2020
ssh: connect to host 192.168.0.202 port 8122: No route to host

[11:43:13.806] Resolver error: Error: No route to host
	at Function.Offline (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:64692)
	at /home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:62497
	at Object.t.handleInstallOutput (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:62620)
	at Object.e [as tryInstallWithLocalServer] (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:386419)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async /home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:294035
	at async Object.t.withShowDetailsEvent (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:405329)
	at async /home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:384890
	at async E (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:381658)
	at async Object.t.resolveWithLocalServer (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:384517)
	at async Object.t.resolve (/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:1:295505)
	at async /home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/extension.js:127:110530
[11:43:13.811] ------



```

**Comment by ICHx** [comment]
The same issue still persist.  Local x64, target arm64 linux

> Other users are able to

**Comment by ICHx** [comment]
Outside vscode 1.55.2
```
 ssh zuk
Welcome to Ubuntu 20.04.2 LTS (GNU/Linux 4.4.205-LineageOS+ aarch64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
Ubuntu 20.04 LTS [running via Linux Deploy]
Last login: Wed May  5 11:29:03 2021 from 192.168.0.119
ZUK:phone@zuk:~$ 
```

**Comment by ICHx** [comment]
Rollback again to 1.55.1
```
[11:49:58.768] Log Level: 2
[11:49:58.771] remote-ssh@0.65.4
[11:49:58.775] linux x64
[11:49:58.777] SSH Resolver called for "ssh-remote+zuk", attempt 1
[11:49:58.777] "remote.SSH.useLocalServer": true
[11:49:58.778] "remote.SSH.path": undefined
[11:49:58.778] "remote.SSH.configFile": undefined
[11:49:58.778] "remote.SSH.useFlock": true
[11:49:58.778] "remote.SSH.lockfilesInTmp": false
[11:49:58.778] "remote.SSH.localServerDownload": auto
[11:49:58.778] "remote.SSH.remoteServerListenOnSocket": false
[11:49:58.778] "remote.SSH.showLoginTerminal": false
[11:49:58.778] "remote.SSH.defaultExtensions": []
[11:49:58.779] "remote.SSH.loglevel": 2
[11:49:58.779] SSH Resolver called for host: zuk
[11:49:58.779] Setting up SSH remote "zuk"
[11:49:58.782] Acquiring local install lock: /tmp/vscode-remote-ssh-zuk-install.lock
[11:49:58.790] Looking for existing server data file at /home/ricts/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-08a217c4d27a02a5bcde898fd7981bda5b49391b-0.65.4/data.json
[11:49:58.791] Using commit id "08a217c4d27a02a5bcde898fd7981bda5b49391b" and quality "stable" for server
[11:49:58.794] Install and start server if needed
[11:49:58.798] PATH: /home/ricts/.local/bin:/home/ricts/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[11:49:58.799] Checking ssh with "ssh -V"
[11:49:58.804] > OpenSSH_8.2p1 Ubuntu-4ubuntu0.2, OpenSSL 1.1.1f  31 Mar 2020

[11:49:58.808] askpass server listening on /run/user/1000/vscode-ssh-askpass-26f5475079f9bf07b0a78c0514d0b1b3c675896f.sock
[11:49:58.808] Spawning local server with {"ipcHandlePath":"/run/user/1000/vscode-ssh-askpass-af143d3851d7c2de43eaab884aaaae676ce8f834.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","36655","-o","ConnectTimeout=15","zuk"],"dataFilePath":"/home/ricts/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-08a217c4d27a02a5bcde898fd7981bda5b49391b-0.65.4/data.json"}
[11:49:58.808] Local server env: {"DISPLAY":":0","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/usr/share/code/code","VSCODE_SSH_ASKPASS_MAIN":"/home/ricts/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.4/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/run/user/1000/vscode-ssh-askpass-26f5475079f9bf07b0a78c0514d0b1b3c675896f.sock"}
[11:49:58.811] Spawned 11276
[11:49:58.890] > local-server> Spawned ssh: 11285
[11:49:58.891] stderr> OpenSSH_8.2p1 Ubuntu-4ubuntu0.2, OpenSSL 1.1.1f  31 Mar 2020
[11:49:59.154] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:2Op7xAXkYvXk9w0mP4jQBBY9s9QTqLRK51CuJ2azRmg
[11:49:59.292] stderr> Authenticated to 192.168.0.202 ([192.168.0.202]:8122).
[11:49:59.578] > Welcome to Ubuntu 20.04.2 LTS (GNU/Linux 4.4.205-LineageOS+ aarch64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> Ubuntu 20.04 LTS [running via Linux Deploy]
[11:49:59.595] > ready: 0c9bdb8744e8
[11:49:59.614] > Linux 4.4.205-LineageOS+ #1 SMP PREEMPT Thu Apr 8 10:51:38 UTC 2021
[11:49:59.614] Platform: linux
[11:49:59.682] > 0c9bdb8744e8: running
[11:49:59.888] > Acquiring lock on /home/phone/.vscode-server/bin/08a217c4d27a02a5bcde898fd7981bda5b49391b/vscode-remote-lock.phone.08a217c4d27a02a5bcde898fd7981bda5b49391b
[11:49:59.903] > Found existing installation at /home/phone/.vscode-server/bin/08a217c4d27a02a5bcde898fd7981bda5b49391b...
> Checking /home/phone/.vscode-server/.08a217c4d27a02a5bcde898fd7981bda5b49391b.log and /home/phone/.vscode-server/.08a217c4d27a02a5bcde898fd7981bda5b49391b.pid for a running server
> Looking for server with pid: 27029
[11:49:59.919] > Starting server with command... /home/phone/.vscode-server/bin/08a217c4d27a02a5bcde898fd7981bda5b49391b/server.sh --start-server --host=127.0.0.1 --enable-remote-auto-shutdown  --port=0  &> "/home/phone/.vscode-server/.08a217c4d27a02a5bcde898fd7981bda5b49391b.log" < /dev/null
> printenv:
[11:49:59.926] >     SHELL=/bin/bash
>     PWD=/home/phone
>     LOGNAME=phone
>     MOTD_SHOWN=pam
>     HOME=/home/phone
>     LANG=C
>     VSCODE_AGENT_FOLDER=/home/phone/.vscode-server
>     SSH_CONNECTION=192.168.0.119 48940 192.168.0.202 8122
>     USER=phone
>     SHLVL=1
>     SSH_CLIENT=192.168.0.119 48940 8122
>     PATH=/home/phone/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/hadoop/bin
>     _=/usr/bin/printenv
[11:49:59.951] > Spawned remote server: 6067
[11:49:59.970] > Waiting for server log...
[11:50:00.017] > Waiting for server log...
[11:50:00.064] > Waiting for server log...
[11:50:00.115] > Waiting for server log...
[11:50:00.167] > Waiting for server log...
[11:50:00.208] > Waiting for server log...
[11:50:00.246] > Waiting for server log...
[11:50:00.287] > Waiting for server log...
[11:50:00.327] > Waiting for server log...
[11:50:00.379] > Waiting for server log...
[11:50:00.422] > Waiting for server log...
[11:50:00.473] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[11:50:00.483] > 0c9bdb8744e8: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==37224==
> osReleaseId==ubuntu==
> arch==aarch64==
> tmpDir==/tmp==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==535==
> connectionToken==1aa111a1-a1a1-1111-111a-a1a11a11111a==
> 0c9bdb8744e8: end
[11:50:00.484] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==37224==
osReleaseId==ubuntu==
arch==aarch64==
tmpDir==/tmp==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime==535==
connectionToken==1aa111a1-a1a1-1111-111a-a1a11a11111a==

[11:50:00.485] Remote server is listening on 37224
[11:50:00.485] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":37224},"osReleaseId":"ubuntu","arch":"aarch64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/tmp","platform":"linux","connectionToken":"1aa111a1-a1a1-1111-111a-a1a11a11111a"},"serverStartTime":535,"installUnpackCode":""}
[11:50:00.485] ** Note: Support for architecture "aarch64" is in preview **
[11:50:00.487] Persisting server connection details to /home/ricts/.config/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-zuk-08a217c4d27a02a5bcde898fd7981bda5b49391b-0.65.4/data.json
[11:50:00.490] Starting forwarding server. localPort 38425 -> socksPort 36655 -> remotePort 37224
[11:50:00.491] Forwarding server listening on 38425
[11:50:00.491] Waiting for ssh tunnel to be ready
[11:50:00.492] [Forwarding server 38425] Got connection 0
[11:50:00.493] Tunneled 37224 to local port 38425
[11:50:00.493] Resolved "ssh-remote+zuk" to "127.0.0.1:38425"
[11:50:00.502] ------




[11:50:00.528] [Forwarding server 38425] Got connection 1
[11:50:01.206] [Forwarding server 38425] Got connection 2
[11:50:03.766] [Forwarding server 38425] Got connection 3

```
